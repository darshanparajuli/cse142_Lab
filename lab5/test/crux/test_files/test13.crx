// test array access

array x : int[4];
array y : int[4][5];
array z : int[3][5][7][9];
var v : int;

func main() : void {
  ::printInt(x);
  ::printInt(y[1]);

  ::printInt(x[1][1]);
  ::printInt(y[1][1][1]);

  ::printInt(z);
  ::printInt(z[1][1]);
  
  let x[1] = 1;
  let v = x[1];

  let x[y[1][2]] = x[y[1][1]];
}
