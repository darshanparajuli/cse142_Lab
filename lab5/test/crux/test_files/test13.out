Error type-checking file.
TypeError(9,3)[Cannot call func(TypeList(int)):void using TypeList(array[4,int]).]
TypeError(10,3)[Cannot call func(TypeList(int)):void using TypeList(array[5,int]).]
TypeError(12,19)[Cannot index Address(int) with int.]
TypeError(12,14)[Cannot dereference ErrorType(Cannot index Address(int) with int.)]
TypeError(12,3)[Cannot call func(TypeList(int)):void using TypeList(ErrorType(Cannot dereference ErrorType(Cannot index Address(int) with int.))).]
TypeError(13,22)[Cannot index Address(int) with int.]
TypeError(13,14)[Cannot dereference ErrorType(Cannot index Address(int) with int.)]
TypeError(13,3)[Cannot call func(TypeList(int)):void using TypeList(ErrorType(Cannot dereference ErrorType(Cannot index Address(int) with int.))).]
TypeError(15,3)[Cannot call func(TypeList(int)):void using TypeList(array[3,array[5,array[7,array[9,int]]]]).]
TypeError(16,3)[Cannot call func(TypeList(int)):void using TypeList(array[7,array[9,int]]).]

